vembrane/ann_types.py

```diff
@@ -3,10 +3,10 @@
 from typing import Union, Iterable, Tuple, Dict, Callable, Any
 import re
 
-from .errors import MoreThanOneAltAllele
+from .errors import MoreThanOneAltAllele, NotExactlyOneValue
 
 
-class NoValue:
+class NoValue(str):
```

Comment:
Can you elaborate on this change?

Reply:
If NoValue inherits from str, `re.search("something", NoValue())` does not error but just come up empty handed, which is convenient behaviour. This way, we do not have to special case / monkey patch / wrap the regex module.

Reply:
I see, thanks.
We do have to come up with another solution though, since now, e.g., `NA == ''`.

Reply:
> I see, thanks.
> We do have to come up with another solution though, since now, e.g., `NA == ''`.

No, `NA == x` is `False` for any `x`.

Reply:
Right, but with the proposed change and due to inheritance we now not only have `isinstance(NA, str) is True` but also `NoValue.__call__ is str.__call__` and `NoValue() == str()`, i.e., `NA.__eq__('')`.

Reply:
(`str.__eq__` is also inherited.)

Reply:
> Right, but with the proposed change and due to inheritance we now not only have `isinstance(NA, str) is True` but also `NoValue.__call__ is str.__call__` and `NoValue() == str()`, i.e., `NA.__eq__('')`.

Not quite:
 `isinstance(NA, str) == True`
`( NoValue.__call__ is str.__call__ ) == False`
`( NoValue() == str() ) == False`
`NA.__eq__('') == False`

Reply:
> (`str.__eq__` is also inherited.)

... but overloaded by

```python
    def __eq__(self, other):
        return False
```

Reply:
Ahhhhhhhhhhhhh!
Collapsed diffs in GitHub.....
Nvm

Reply:
> If NoValue inherits from str, `re.search("something", NoValue())` does not error but just comes up empty handed, which is convenient behaviour. This way, we do not have to special case / monkey patch / wrap the regex module.

Can you add this as a comment to record the intention for posterity? :)

Reply:
Will do, will also update the README